src/util.h

```diff
@@ -341,15 +341,30 @@ class MaybeStackBuffer {
     AllocateSufficientStorage(storage);
   }
 
+  bool IsAllocated() {
+    return buf_ != buf_st_;
+  }
+
   ~MaybeStackBuffer() {
-    if (buf_ != buf_st_)
+    // Note: if Release() is called, this memory will not be freed here.
+    if (!released_ && buf_ != buf_st_)
       free(buf_);
   }
 
+  // Caution: Causes allocations to not be freed when instances
+  // are deconstructed. By calling Release, ownership (and responsibility
+  // for) the allocated internal buffer can be transferred to another
+  // object (like a Buffer) without copying.
+  bool Release() {
+    released_ = IsAllocated();
+    return released_;
+  }
+
  private:
   size_t length_;
   T* buf_;
   T buf_st_[kStackStorageSize];
+  bool released_;
```

Comment:
Yep.. would need to make sure Release() is called after (instead of before) like I have it now, but this is goodness.